### PR TITLE
Fix const override

### PR DESCRIPTION
**Pull request by dreamer-dead**
Fix for wrong functions override and tons of compiler warnings.
Without const some methods will be hidden but not overloaded.

During this work I moved generic implementation for Counters texture generation.
In future it should be better to use override\const override instead of just virtual.

### CODE REVIEW

File: src/Graphics/UI.h [hunk 1]
```diff
@@ -75,7 +75,7 @@ class UI
     int _y = 0;
     int _xOffset = 0;
     int _yOffset = 0;
-    Texture* _texture = 0;
+    mutable Texture* _texture = 0;
```
Comment (alexeevdv) [inline src/Graphics/UI.h]: wow wow wow... what is this for?

**Comment by alexeevdv** [comment]
**override** sprecifier is new for me. I'll try to use it in future. Looks like it is more convinient than **virtual**

File: src/Graphics/Texture.cpp [hunk 2]
```diff
@@ -306,4 +306,36 @@ bool Texture::blitWithAlpha(Texture* blitMask, int maskOffsetX, int maskOffsetY)
...
+    unsigned int maxLength,
+    Texture* symbolSource,
+    unsigned int charWidth,
+    unsigned int charHeight,
+    unsigned int xOffsetByColor,
+    bool isSigned)
+{
+    // number as text
+    const auto charsCount = maxLength + (isSigned ? 1 : 0);
+    auto texture = std::unique_ptr<Texture>(new Texture(charWidth * charsCount, charHeight));
```
Comment (alexeevdv) [inline src/Graphics/Texture.cpp]: We can define std::make_unique according to http://herbsutter.com/gotw/_102/ and use it in our code. What do you think about it?

Comment (dreamer-dead) [inline src/Graphics/Texture.cpp]: Unfortunately not now (
It's c++14 featute and my STL library doesn't support it.

Comment (alexeevdv) [inline src/Graphics/Texture.cpp]: I mean that we can define our own std::make_unique function which can be replaced in future when C++14 is ready

File: src/Graphics/UI.h (hunk 1 above)
Comment (dreamer-dead) [inline src/Graphics/UI.h]: I made it mutable to have a const texture() getter.
It is used in width() and height() getters and that's why it should be const.
I understand that this solution isn't perfect (

File: src/Graphics/Texture.cpp (hunk 2 above)
Comment (dreamer-dead) [inline src/Graphics/Texture.cpp]: Yes, we can.
It's not a very complex feature.

File: src/Graphics/UI.h (hunk 1 above)
Comment (phobos2077) [inline src/Graphics/UI.h]: Why you can't have const texture() getter without "mutable" on the _texture?

Comment (alexeevdv) [inline src/Graphics/UI.h]: Because in some inherited classes _texture object is created in texture() method. **Const** specifier don't allow this

Comment (dreamer-dead) [inline src/Graphics/UI.h]: It's a specific to inherited classes but not to UI class.
And there can be a better solution that affects only classes like BigCounter.
I'll spend some more time on that.

**Comment by dreamer-dead** [comment]
I added make_unique helper.
Put it in our namespace and not in std.
We always can add line 'using std::make_unique;' to fix it when c++14 became available.

**Comment by Oppen** [comment]
Why not using namespace std?

**Comment by dreamer-dead** [comment]
Oppen@, 
http://stackoverflow.com/questions/320798/adding-types-to-the-std-namespace

[C++11: 17.6.4.2.1/1]: The behavior of a C++ program is undefined if it adds declarations or definitions to namespace std or to a namespace within namespace std unless otherwise specified.

We will break compilation with c++14 compliant compiler and standard library.
Or we should detect that and do some additional work to avoid collisions.

**Comment by vvbalashoff** [comment]
please rename make_unique to makeUnique

**Comment by alexeevdv** [comment]
@vvbalashoff, make_unique is library function from C++14. There is no need to change it name

**Comment by dreamer-dead** [comment]
Here is my attempt to avoid mutable keyword in base Texture class.
But instead I was forced to add new field for on-demand texture in classes SmallCounter and BigCounter.

**Pull request merged**